subprojects/packagefiles/libarchive/meson.build

```diff
@@ -2,7 +2,7 @@ project(
   'libarchive',
   'c',
   version: '3.6.2',
-  meson_version: '>=0.49.0',
```

Comment:
Unwritten rule is to keep compatibility with meson 0.49, which is included with Debian 9. I don't see a compelling reason to increase to 0.60.

Reply:
This really should be mentioned in the "How to contribute new wraps" section…

Reply:
One of the reasons it's unwritten is because wraps aren't really tested on Debian 9. Last time I did that, I found that meson 0.49 crashes with ternary operators (?: instead of if else). Another reason is that some wraps really need a newer version.

Anyway, 0.60 is really new. I don't think most OSes include it.

Reply:
That does not make any sense. Either it's a rule, and it should be enforced by running the CI with the minimum supported version, or it's just a lie.

Anyway, even without the iconv change, it does not work with `0.49` (`disabler().partial_dependency(…)` does not work), would need at least `0.54` anyway (for the former issue and `meson.override_dependency`).

I don't think it's too much to ask for a semi-recent version of meson.

Reply:
It's not even a rule, really.

But it is a good rule of *thumb*, that wraps should work on the oldest version of Meson that it isn't too painful to support. I'm not a fan of people marking wraps as requiring very recent Meson "just because that is all I tested it with", and sometimes it's very easy to support old versions with low effort.

The right balance is probably "review comment: you could probably avoid bumping the meson_version so high by just doing XXX. I would do so, but it's up to you in the end".

Reply:
0.49.0 is a very robust checkpoint because it unlocks use by people running Debian oldstable using apt instead of pip to install meson.

0.56.0 is another good checkpoint because Debian stable has it.

Reply:
Unclear guidelines waste both our time. It may not be painful to support older versions, but it does take more time. At the very least all wraps should be updated with `meson_version: '>=0.49.0',` so you get a warning when using a feature only provided by a more recent version.